File: CoreServices/src/main/java/com/example/entities/Review.java
package com.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "review")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Review {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    private int rating;
    @Lob
    @Column(columnDefinition = "TEXT")
    private String reviewText;
    private LocalDateTime timestamp;
    @ManyToOne
    @JoinColumn(name = "appointment_id", nullable = false)
    private Appointment appointment;
    @ManyToOne
    @JoinColumn(name = "doctor_id", nullable = false)
    private Doctor doctor;
    @ManyToOne
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;

    @PrePersist
    protected void onCreate() {
        timestamp = LocalDateTime.now();
    }
}

File: GUI/src/app/shared/models/review/review.ts
import { v4 as uuidv4 } from 'uuid';
import { Doctor } from '../users/doctor/doctor';
import { Patient } from '../users/patient/patient';

export class Review {
    id?: string;
    rating: number;
    reviewText: string;
    timestamp: Date;
    appointmentId: string;
    doctor: Doctor;
    patient: Patient;

    constructor() {
        this.id = uuidv4();
    }
}
